src/ResourceManager/AzureBatch/Commands.Batch/ComputeNodes/StartBatchComputeNodeServiceLogUploadCommand.cs

```diff
@@ -0,0 +1,73 @@
+﻿// ----------------------------------------------------------------------------------
+//
+// Copyright Microsoft Corporation
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// http://www.apache.org/licenses/LICENSE-2.0
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// ----------------------------------------------------------------------------------
+
+using System;
+using System.Linq;
+using Microsoft.Azure.Commands.Batch.Models;
+using System.Management.Automation;
+using Constants = Microsoft.Azure.Commands.Batch.Utils.Constants;
+
+namespace Microsoft.Azure.Commands.Batch
+{
+    [Cmdlet(VerbsLifecycle.Start, 
+        Constants.AzureBatchComputeNodeServiceLogUpload,
+        SupportsShouldProcess = true,
+        DefaultParameterSetName = Constants.AzureBatchComputeNodeServiceLogUpload),         
+         OutputType(typeof(PSStartComputeNodeServiceLogUploadResult))]
+    public class StartBatchComputeNodeServiceLogUploadCommand : BatchObjectModelCmdletBase
+    {
+        [Parameter(Position = 0, ParameterSetName = Constants.IdParameterSet, Mandatory = true,
+            HelpMessage = "The id of the pool that contains the compute node.")]
+        [ValidateNotNullOrEmpty]
+        public string PoolId { get; set; }
+
+        [Parameter(Position = 1, ParameterSetName = Constants.IdParameterSet, Mandatory = true,
+            HelpMessage = "The id of the compute node.")]
+        [ValidateNotNullOrEmpty]
+        public string ComputeNodeId { get; set; }
+
+        [Parameter(Position = 1, ParameterSetName = Constants.ParentObjectParameterSet, ValueFromPipeline = true)]
```

Comment:
@antoniowmsft I believe this should be mandatory in its given parameter set

Reply:
Done.